outputs/otlp.go

```diff
@@ -0,0 +1,172 @@
+	c.Stats.OTLP.Add(Total, 1)
+
+	trace := c.newTrace(falcopayload)
+	if trace == nil {
+		go c.CountMetric(Outputs, 1, []string{"output:otlp/traces", "status:error"})
+		c.Stats.OTLP.Add(Error, 1)
+		c.PromStats.Outputs.With(map[string]string{"destination": "otlp/traces", "status": Error}).Inc()
+		log.Printf("[ERROR] : OLTP Traces - Error generating trace")
+		return
+	}
+	// Setting the success status
+	go c.CountMetric(Outputs, 1, []string{"output:otlp/traces", "status:ok"})
+	c.Stats.OTLP.Add(OK, 1)
+	c.PromStats.Outputs.With(map[string]string{"destination": "otlp/traces", "status": OK}).Inc()
+}
+
+const (
+	templateOption       = "missingkey=zero"
+	kubeTemplateStr      = `{{.cluster}}{{.k8s_pod_name}}{{.k8s_ns_name}}{{.k8s_container_name}}`
+	containerTemplateStr = `{{.container_id}}`
+)
+
+var (
+	kubeTemplate      = template.Must(template.New("").Option(templateOption).Parse(kubeTemplateStr))
+	containerTemplate = template.Must(template.New("").Option(templateOption).Parse(containerTemplateStr))
+)
+
+func sanitizeOutputFields(falcopayload types.FalcoPayload) map[string]interface{} {
+	ret := make(map[string]interface{})
+	for k, v := range falcopayload.OutputFields {
+		k := strings.ReplaceAll(k, ".", "_")
+		ret[k] = v
+	}
+	return ret
+}
+
+func renderTraceIDFromTemplate(falcopayload types.FalcoPayload, config *types.Configuration) (string, string) {
+	tplStr := config.OTLP.Traces.TraceIDHash
+	tpl := config.OTLP.Traces.TraceIDHashTemplate
+	outputFields := sanitizeOutputFields(falcopayload)
+	if tplStr == "" {
+		switch {
+		case outputFields["cluster"] != nil &&
+			outputFields["k8s_pod_name"] != nil &&
+			outputFields["k8s_ns_name"] != nil &&
+			outputFields["k8s_container_name"] != nil:
+			tpl, tplStr = kubeTemplate, kubeTemplateStr
+		default:
+			tpl, tplStr = containerTemplate, containerTemplateStr
+		}
+	}
+	buf := &bytes.Buffer{}
+	if err := tpl.Execute(buf, outputFields); err != nil {
+		log.Printf("[WARNING] : OTLP - Error expanding template: %v", err)
+	}
+	return buf.String(), tplStr
+}
+
+func generateTraceID(falcopayload types.FalcoPayload, config *types.Configuration) (trace.TraceID, string, error) {
+	var traceID trace.TraceID
+	var err error
+	traceIDStr, tplStr := renderTraceIDFromTemplate(falcopayload, config)
+
+	switch traceIDStr {
+	case "":
+	case "<no value>":
+		// Template produced no string :(, generate a random 32 character string
+		traceIDStr, err = randomHex(16)
+		if err != nil {
+			return traceID, "", err
+		}
+	default:
+		// Hash the returned template- rendered string to generate a 32 character traceID
+		hash := fnv.New128a()
+		hash.Write([]byte(traceIDStr))
+		digest := hash.Sum(nil)
+		traceIDStr = hex.EncodeToString(digest[:])
+	}
+	traceID, err = trace.TraceIDFromHex(traceIDStr)
+	return traceID, tplStr, err
+}
+
+func randomHex(n int) (string, error) {
+	bytes := make([]byte, n)
+	if _, err := rand.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
```

Comment:
To generate an UUID for each event, I'm using [github.com/google/uuid](https://github.com/google/uuid). See https://github.com/falcosecurity/falcosidekick/blob/5cfd355809f38d990b5458f354b091a653af7798/handlers.go#L102

It could replace this whole function and have a simpler code, wdyt?

Reply:
Well nopes directly, I tried and failed, as the traceID needs to be _exactly_ 32chars hexa.
That said, we could re-use that UUID and massage it (remove the `-`, and cut it from 36chars to 32chars).
Opinion?

Reply:
hm actually, it was indeed pretty easy to just use UUID string as "seed" for the traceID in this case (no value from template rendering), implemented at a45a91da